src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,119 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private boolean checkIsEmpty;
+    private T[] array;
+
+    public ArrayList() {
+        checkIsEmpty = true;
+        array = (T[]) new Object[2];
+    }
+
+    @Override
+    public void add(T value) {
+        checkIsEmpty = false;
+        if (position < array.length) {
+            array[position] = value;
+            position++;
+        } else {
+            int newArrayLength = array.length + (array.length >> 1);
+            array = Arrays.copyOf(array, newArrayLength);
+            array[position] = value;
+            position++;
+        }
+    }
+
+    @Override
+    public void add(T value, int index) {
+        checkIsEmpty = false;
+        if (index < array.length && index > -1) {
+            array[index] = value;
+        }
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        checkIsEmpty = false;
+        int newArrayLength = array.length + list.size();
+        array = Arrays.copyOf(array, newArrayLength);
+
+        for (int i = 0, j = 0; i < array.length; i++) {
+            if (array[i] == null) {
+                array[i] = list.get(j);
+                j++;
+            }
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        if (index < array.length && index > -1 && array[index] != null) {
+            return array[index];
+        }
+        return null;
+    }
+
+    @Override
+    public void set(T value, int index) {
+        checkIsEmpty = false;
+        if (index < array.length && index > -1) {
+            array[index] = value;
```

Comment:
list = new ArrayList();
list.set(2, 2);
test what happens here?
this method replaces the element at the specified position in this list with the specified element, doesn't add new element

Reply:
You are right!